org.scala-ide.sbt.core/src/org/scalaide/sbt/core/SbtBuild.scala

```diff
@@ -59,6 +67,22 @@ object SbtBuild extends AnyRef with HasLogger {
     }
   }
 
+  def shutdown(implicit system: ExecutionContext): Future[Boolean] = Future.sequence {
+    builds.values
+  }.flatMap { builds =>
+    buildsLock.synchronized {
+      builds.foreach { _.shutdown() }
+    }
+    @tailrec def areClosed: Boolean = if (builds.forall { _.isClosed })
```

Comment:
I think this polling is too expensive. We should put in some `Thread.sleep(100)`. Or use an akka scheduler to rerun the test continually.